packages/matter.js/test/behavior/definitions/window-covering/WindowCoveringServerTest.ts

```diff
@@ -0,0 +1,98 @@
+ */
+
+import { WindowCoveringServer } from "../../../../src/behavior/definitions/window-covering/WindowCoveringServer.js";
+import { WindowCovering } from "../../../../src/cluster/definitions/WindowCoveringCluster.js";
+import { MockEndpoint } from "../../../endpoint/mock-endpoint.js";
+
+class TestWindowCoveringServer extends WindowCoveringServer.with(
+    "Lift",
+    "Tilt",
+    "PositionAwareLift",
+    "PositionAwareTilt",
+).set({
+    type: WindowCovering.WindowCoveringType.TiltBlindLift,
+}) {
+    override initialize() {
+        this.state.currentPositionLiftPercent100ths = 5000; // Half open
+        this.state.currentPositionTiltPercent100ths = 5000; // Half tilted
+
+        return super.initialize();
+    }
+
+    override async handleMovement() {
+        // Skip default logic which is just an instant "jump" to target position and thus makes operationalStatus
+        // changes uninteresting
+    }
+}
+
+async function createTiltBlindLift() {
+    return MockEndpoint.createWith(TestWindowCoveringServer);
+}
+
+describe("WindowCoveringServer", () => {
+    it("emits operationalStatus change events when target position changes", async () => {
+        const device = await createTiltBlindLift();
+
+        const events = device.captureEvents(WindowCoveringServer, {
+            names: ["operationalStatus$Changing", "operationalStatus$Changed"],
+        });
+
+        await device.act(async agent => await agent.windowCovering.downOrClose());
+
+        await device.act(async agent => await agent.windowCovering.upOrOpen());
+
+        await device.act(async agent => await agent.windowCovering.stopMotion());
+
+        expect(events).deep.equals([
+            {
+                name: "operationalStatus$Changing",
+                oldValue: { lift: 0, tilt: 0, global: 0 },
+                newValue: { global: 2, lift: 2, tilt: 2 },
+            },
+            {
+                name: "operationalStatus$Changing",
+                oldValue: { lift: 2, tilt: 2, global: 0 },
+                newValue: { global: 2, lift: 2, tilt: 2 },
+            },
+            {
+                name: "operationalStatus$Changed",
+                oldValue: { lift: 0, tilt: 0, global: 0 },
+                newValue: { lift: 2, tilt: 2, global: 2 },
+            },
+            {
+                name: "operationalStatus$Changing",
+                oldValue: { lift: 2, tilt: 2, global: 2 },
+                newValue: { global: 1, lift: 1, tilt: 1 },
+            },
+            {
+                name: "operationalStatus$Changing",
+                oldValue: { lift: 1, tilt: 1, global: 2 },
+                newValue: { global: 1, lift: 1, tilt: 1 },
+            },
+            {
+                name: "operationalStatus$Changed",
+                oldValue: { lift: 2, tilt: 2, global: 2 },
+                newValue: { lift: 1, tilt: 1, global: 1 },
+            },
+            {
+                name: "operationalStatus$Changed",
```

Comment:
Any idea why the orderhere is that way around? Changed first and afterwards the Changing?

Reply:
Off the top of my head the best idea I have is "WTF?"

Reply:
Was a bug in ObservableProxy, good catch!